Some vanilla commands prevent mcfunctions from being loaded

### Expected behavior

Commands like `msg Notch hi` don't prevent a mcfunction from loading.

### Observed/Actual behavior

Adding `msg Notch hi` to a .mcfunction file results in an exception:

[12:24:21 ERROR]: Failed to load function minecraft:errortest/errortest
java.util.concurrent.CompletionException: java.lang.UnsupportedOperationException: Not supported yet.
        at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314) ~[?:?]
        at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:319) [?:?]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1766) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
        at java.lang.Thread.run(Thread.java:831) [?:?]
Caused by: java.lang.UnsupportedOperationException: Not supported yet.
        at net.minecraft.commands.ICommandListener$1.getBukkitSender(ICommandListener.java:30) ~[patched_1.17.1.jar:git-Paper-218]
        at net.minecraft.commands.CommandListenerWrapper.getBukkitSender(CommandListenerWrapper.java:322) ~[patched_1.17.1.jar:git-Paper-218]
        at net.minecraft.commands.CommandDispatcher.lambda$new$0(CommandDispatcher.java:210) ~[patched_1.17.1.jar:git-Paper-218]
        at com.mojang.brigadier.tree.CommandNode.canUse(CommandNode.java:78) ~[patched_1.17.1.jar:git-Paper-218]
        at com.mojang.brigadier.CommandDispatcher.parseNodes(CommandDispatcher.java:358) ~[patched_1.17.1.jar:git-Paper-218]
        at com.mojang.brigadier.CommandDispatcher.parse(CommandDispatcher.java:348) ~[patched_1.17.1.jar:git-Paper-218]
        at net.minecraft.commands.CustomFunction.a(CustomFunction.java:61) ~[patched_1.17.1.jar:git-Paper-218]
        at net.minecraft.server.CustomFunctionManager.lambda$reload$3(CustomFunctionManager.java:86) ~[patched_1.17.1.jar:git-Paper-218]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1764) ~[?:?]
        ... 3 more

Namespacing doesn't work (since functions don't support plugin commands in the first place), and not all vanilla commands are affected (for example `gamemode creative Notch` works fine).

### Steps/models to reproduce

Add a new datapack to a world, create a test.mcfunction file, and add `msg Notch hi` to it. You can also use the datapack provided here and simply add it to the world/datapacks folder. Start the server.

[errortest.zip](https://github.com/PaperMC/Paper/files/7056161/errortest.zip)


### Plugin list

None

### Paper version

> version
[12:24:30 INFO]: Checking version, please wait...
[12:24:31 INFO]: This server is running Paper version git-Paper-218 (MC: 1.17.1) (Implementing API version 1.17.1-R0.1-SNAPSHOT) (Git: e763a3d)
You are running the latest version
Previous version: git-Paper-216 (MC: 1.17.1)

This has been happening since version 216, so [this commit](https://github.com/PaperMC/Paper/commit/3773df24ce81173242bff99a2a9991f7a88a66b2) is most likely the cause.

### Agreements

- [X] I am running the latest version of Paper available from https://papermc.io/downloads.
- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] My version of Minecraft is supported by Paper.

### Other

_No response_

--- a/patches/server/0745-Vanilla-command-permission-fixes.patch
+++ b/patches/server/0745-Vanilla-command-permission-fixes.patch
@@ -30,7 +30,7 @@ index 899008b2980d13f1be6280cd8cb959c53a29bebf..f875507241ac6769545e91cd3285232b
      private RedirectModifier<S> modifier = null;
      private boolean forks;
 diff --git a/src/main/java/net/minecraft/commands/Commands.java b/src/main/java/net/minecraft/commands/Commands.java
-index 816ea85a5880123c970d227d61426a0ae2b9539f..6fec6a47538da4c0c5a4505e9bedf492bb3376dd 100644
+index 816ea85a5880123c970d227d61426a0ae2b9539f..b7c24a7ff4dc50d03f74b21a0b353d0ce559b4d4 100644
 --- a/src/main/java/net/minecraft/commands/Commands.java
 +++ b/src/main/java/net/minecraft/commands/Commands.java
 @@ -204,6 +204,13 @@ public class Commands {
@@ -40,7 +40,7 @@ index 816ea85a5880123c970d227d61426a0ae2b9539f..6fec6a47538da4c0c5a4505e9bedf492
 +        // Paper start
 +        for (final CommandNode<CommandSourceStack> node : this.dispatcher.getRoot().getChildren()) {
 +            if (node.getRequirement() == com.mojang.brigadier.builder.ArgumentBuilder.<CommandSourceStack>defaultRequirement()) {
-+                node.requirement = stack -> stack.getBukkitSender().hasPermission(org.bukkit.craftbukkit.command.VanillaCommandWrapper.getPermission(node));
++                node.requirement = stack -> stack.source == CommandSource.NULL || stack.getBukkitSender().hasPermission(org.bukkit.craftbukkit.command.VanillaCommandWrapper.getPermission(node));
 +            }
 +        }
 +        // Paper end

